### ISSUE TITLE
diff on multi-library ref packages reports 'no changes' without --platform

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#293]
## Summary

When using `diff --package Microsoft.NETCore.App.Ref@P2..P3`, the command reports `no changes` even though there are significant API additions across multiple libraries.

## Steps to reproduce

```bash
# This reports 'no changes'
dnx dotnet-inspect -y -- diff \
  --package "Microsoft.NETCore.App.Ref@11.0.0-preview.2.26159.112..11.0.0-preview.3.26168.106" \
  --source "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet11/nuget/v3/index.json"

# But this correctly finds the new ProcessStartOptions type
dnx dotnet-inspect -y -- diff \
  --platform "System.Diagnostics.Process@11.0.0-preview.2.26159.112..11.0.0-preview.3.26168.106" \
  --source "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet11/nuget/v3/index.json" \
  --additive
```

## Expected behavior

`diff --package Microsoft.NETCore.App.Ref@P2..P3` should aggregate changes across all libraries in the ref package, or at minimum warn that it's only comparing one library.

## Workaround

Using `--platform <LibraryName>@P2..P3` per-library works correctly. But this requires knowing which libraries changed — you need an external mechanism (like diffing ref source files in the VMR) to identify which libraries to check.

## Additional observations

1. `library --package Microsoft.NETCore.App.Ref@version` returns `No DLLs found in package` — ref packages contain ref assemblies but the library command doesn't find them
2. `member <Type> --package Microsoft.NETCore.App.Ref@version` can't resolve types, but `find <Type> --package ...` can search across all libraries
3. `package Microsoft.NETCore.App.Ref@version -S Files` returns empty

## Use case

We're building a release notes generation skill that needs to diff all public API changes between .NET preview releases. Being able to enumerate libraries in a ref package and diff them would be very valuable.

## Environment

- dotnet-inspect via `dnx`
- macOS arm64
- Feed: dotnet11 Azure DevOps public feed
